### PR TITLE
Change: Use React Query to replace useRegions hook

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

We can't entirely remove Regions from Redux yet, because of several class components that still use the container. However, there's a nice `placeholderData` that makes this a fun case for react-query, so for now I've removed useRegions.ts and replaced it with /queries/regions.

All affected locations should work as before.

I started on removing the usage of regions.container that we're able to do, but that involves some CMR cleanup and would make the PR much bigger so I'm putting this portion up for review.

I removed some error handling logic, see comment inline. I like the way this works with the placeholderData; if the request fails three times in the background, we'll just stick with the cached data (which is stale but for almost all cases is totally fine). Please let me know what you think about this.

### CODE REVIEW

File: packages/manager/src/features/linodes/MigrateLanding/MigrateLinode.tsx [hunk 1]
```diff
@@ -152,17 +150,7 @@ const MigrateLanding: React.FC<CombinedProps> = props => {
...
-  }
-
-  if (regions.error) {
-    return (
-      <Error errorText="There was an issue loading configuration options." />
-    );
-  }
-
-  if (regions.entities.length === 0 && regions.lastUpdated !== 0) {
+  if (regions.length === 0) {
```
Comment (Jskobos) [inline packages/manager/src/features/linodes/MigrateLanding/MigrateLinode.tsx:153]: Removed this stuff. There's a funny bug in production if you block requests to /regions and try to migrate a Linode:

Screen Shot 2021-01-19 at 1 06 12 PM

We could move these states inside the Dialog where they belong, but the caching we do for regions means that even if the request fails, this modal will work fine with slightly stale data.

**Review by DevDW — COMMENTED** [review]
Things are looking good when the request succeeds, but when I block requests to `*regions*` in DevTools, it seems that the intended effect of `placeholderData` isn't there. 

If I'm understanding the purpose of `placeholderData` correctly, when I block that request and go to, for example, Object Storage > Create a Bucket, the region list there should be populated with cached data, but it's showing up with "No results" for me. But the cached data is stored appropriately from what I'm seeing in the React Query DevTool.

(I do think this is a good use of `placeholderData` though since the regions list changes infrequently)

**Comment by Jskobos** [comment]
> Things are looking good when the request succeeds, but when I block requests to `*regions*` in DevTools, it seems that the intended effect of `placeholderData` isn't there.

Good call, I used `initialData` initially (🙃 ) but the docs kind of warned me away from it. Given this though it seems like the better choice for this particular endpoint. See if it matches your intuitions about how this should work. (https://react-query.tanstack.com/guides/initial-query-data)

File: packages/manager/src/mocks/serverHandlers.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { rest, RequestHandler } from 'msw';
 
 import {
-  abuseTicketNotificationFactory,
+  // abuseTicketNotificationFactory,
```
Comment (DevDW) [inline packages/manager/src/mocks/serverHandlers.ts:4]: I think this file was mistakenly committed

**Review by DevDW — APPROVED** [review]
Functionality LGTM now, and using `initialData` allows for the behavior I had expected from `placeholderData` 👍

**Review by johnwcallahan — COMMENTED** [review]
To me it seems `placeholderData` is more appropriate, since with `initialData` the "real" data (from the API request) is never shown. 

This means if (for example) you use the regions query in RegionStatusBanner instead of Redux, DC outages won't be shown. You can simulate this using request overrides in Safari.

To handle for the scenario @DevDW mentioned, you could do something like this in the useQuery hook:

```ts
placeholderData: cachedData as Region[],
onError: () => {
  queryClient.setQueryData('regions', cachedData as Region[]);
}
```

It seems to work but we might want to look at it closer since it seems kind of anti-pattern-y to me (and there might be a better way to achieve this).

**Review by johnwcallahan — APPROVED** [review]
I guess a side effect of using placeholderData is that there's a double /regions request until we move all the region-dependent stuff to RQ, which seems acceptable to me (it's disk cached)

**Pull request merged**